aten/src/ATen/native/cpu/layer_norm_kernel.cpp

```diff
@@ -109,15 +126,32 @@ void LayerNormBackwardKernelImplInternal(
     std::memset(dbeta_data, 0, N * sizeof(T));
   }
   const T scale = T(1) / static_cast<T>(N);
+  constexpr int64_t VEC_SIZE = vec256::Vec256<T>::size();
+  std::array<T, VEC_SIZE> ds_arr;
+  std::array<T, VEC_SIZE> db_arr;
   const bool gamma_null = gamma_data == nullptr;
   for (int64_t i = 0; i < M; ++i) {
     const T* dY_ptr = dY_data + i * N;
     const T* X_ptr = X_data + i * N;
     if (dX_data != nullptr) {
       T* dX_ptr = dX_data + i * N;
-      T ds = 0;
-      T db = 0;
-      for (int64_t j = 0; j < N; ++j) {
+      vec256::Vec256<T> ds_vec(0);
+      vec256::Vec256<T> db_vec(0);
+      int64_t j = 0;
```

Comment:
ditto.

Reply:
The declaration outside loop can keep the value of j in the next loop (See line 168).

Reply:
Actually there is no gain at all in modern compiler but lead to worse readability.

Reply:
Agreed: there is little gain on the modern compiler. In terms of readability, I feel it should have better readability as it avoids the duplication and carries the value of j to the next loop. Check https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cpu/vec256/functional.h#L30-L36 for a similar code piece (`d` is defined before the loop and the value of `d` is carried to the later `if` statement).